Assets/_System/Scripts/TurnBasedScripts/CombatMoverSelector.cs

```diff
@@ -0,0 +1,114 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class CombatMoverSelector : MonoBehaviour
+{
+    public GameObject m_MoveLocationPrefab;
+    public GameObject m_AttackLocationPrefab;
+
+    protected GridObject _movingPiece;
+    protected CombatUnitSelector _unitSelector;
+    protected List<Vector2Int> _moveLocations;
+    protected List<GameObject> _locationHighlights;
+
+    // Start is called before the first frame update
+    protected void Start()
+    {
+        _unitSelector = GetComponent<CombatUnitSelector>();
+        enabled = false;
+    }
+
+    protected void Update()
```

Comment:
This method is a bit long. Could we possibly break it up into smaller easier to understand methods?